adding a PySyft Installation guide to the examples folder

AlanAboudib opened the pull request:
I created a jupyter notebook to provide installation instruction of PySyft for Linux, macOS and Windows.

alaaboudib commented:
@iamtrask @LaRiffle @robert-wagner

robert-wagner commented:
Hey, would it be possible for you to put this in a .md file

AlanAboudib commented:
Sure!

AlanAboudib commented:
Done @robert-wagner

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+<font color='red'>$</font>  `python --version`
+
+#### 2. Install pytorch 1.0:
```
You dont need to install pytorch seperately. It is a dependency

AlanAboudib replied:
The thing is that the exact install command adapted to your  system might change as a function of what option you choose here? https://pytorch.org/get-started/locally/

I am not sure how `pip3 install -r requirements.txt` can handle this??

AlanAboudib replied:
Done!

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+#### 5. Install PySyft
+
+<font color='red'>$</font> `sudo python setup.py install`
```
this probably should not be done with sudo, additionally, with that in mind, pip install pysyft is almost always the better option as it runs this but also can take advantage of wheel files.

AlanAboudib replied:
Done!
Actually, I had to use `sudo` because I am not inside a virtual env. I could have used `--user` installation option too. So yeah, I will remove this and leave it to the user to deal with their permissions

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+<font color='red'>$</font> `brew install python3`
+
+#### 2. Install pytorch 1.0
```
Ditto on not needing to install pytorch

AlanAboudib replied:
Done!

AlanAboudib commented:
You mean in the same folder as the main README.md file?

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+Here is the download link for Git on windows: https://git-scm.com/download/win
+
+#### 5. Install Microsoft Build tools
```
Do you actually need the build tools? Full disclaimer, I do not currently have a windows system to test this on

AlanAboudib replied:
It is actually necessary to get the C++ compiler to my understanding. I couldn't install PySyft without this step

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+Here are the PySyft installation instructions for Linux, MacOs and Windows.
+
+Just scroll down to find your preferred OS:
```
Maybe put links here to the sections of the document rather than saying to scroll down

AlanAboudib replied:
Good idea, done!

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+You can check out the version number by running:
+
+<font color='red'>$</font>  `python --version`
```
You might want to put a link here for instructions on upgrading you python version. It can be a pain but this doc should not be a python install doc :slightly_smiling_face:

AlanAboudib replied:
I totally agree

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+In the terminal
+
+#### 4. Install PySyft
```
It seems like the install instructions are os agnostic (as they should be) maybe move this out seperately for conciseness

AlanAboudib replied:
That is a reasonable idea. However, I thought that each OS section should be complete: A user would expect it to be rather than looking for later sections on the page to complete  their installation procedure. 

While it is redundant from the point of view of the authors, it is not from the point of view of a reader who would most likely target only one section and ignore the others.

What are your thoughts about this @robert-wagner

robert-wagner replied:
That is a good point.

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+## 0.3 Windows
+------------------------------
+
+#### 1. install python
```
Capitalize install and python

AlanAboudib replied:
Done!

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+Here is the link to install python https://www.python.org/downloads
+
+#### 2. add ***python*** and ***pip*** to the `PATH` environment variable
```
Capitalize add

AlanAboudib replied:
Done!

robert-wagner commented on examples/tutorials/Part 0 - Installation Guide.md:
```diff
@@ -0,0 +1,228 @@
+
+Click on `OK`.
+
+#### 3. Install pytorch 1.0
```
Pytorch is a dependency

robert-wagner commented:
Yeah, that is the first place I would look. Also maybe link it from the readme like we have with contributing.md

iamtrask commented:
Very grateful for this contribution!

might need to add "pip install -r requirements.txt" as well just before "python setup.py install"

AlanAboudib commented:
@iamtrask I added requirement installation, and moved the file to the PySyft root folder. I referenced the file from the 'installation' section in the  main README.md file as @robert-wagner recommended.

AlanAboudib commented:
Oh, I haven't seen your comment @robert-wagner , I am checking that out now

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+# PySyft Installation Guide
```
nit: INSTALLATION.md

AlanAboudib replied:
Done!

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+in the terminal. Then, run this to install the PySyft package after installing its requirements:
+
+<font color='red'>$</font> `pip3 install -r requirements.txt`
```
I know @iamtrask made a comment about this but this is not necessary. Setup.py handles this automatically

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+<font color='red'>$</font>  `python --version`
+
+#### 2. Install pytorch 1.0:
```
This isnt need, setup.py takes care of it

AlanAboudib replied:
handled!

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+#### 5. Install PySyft requirements
+
+<font color='red'>$</font> `sudo pip3 install -r requirements.txt`
```
this isnt need. See below

AlanAboudib replied:
Done!

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+#### 6. Install PySyft
+
+<font color='red'>$</font> `sudo python setup.py install`
```
dont run this with sudo

AlanAboudib replied:
Done!

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+#### 7. Test your installation:
+
+<font color='red'>$</font> `sudo python setup.py test`
```
dont run this with sudo

AlanAboudib replied:
done

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+<font color='red'>$</font> `brew install python3`
+
+#### 2. Install pytorch 1.0
```
setup.py automatically installs pytorch with pip

AlanAboudib replied:
handled!

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+## 0.3 Windows
+------------------------------
+
+#### 1. install python
```
Captitalize install

AlanAboudib replied:
done

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+Here is the link to install python https://www.python.org/downloads
+
+#### 2. add ***python*** and ***pip*** to the `PATH` environment variable
```
Capitalize add

AlanAboudib replied:
done

robert-wagner commented on installation_guide.md:
```diff
@@ -0,0 +1,237 @@
+
+Click on `OK`.
+
+#### 3. Install pytorch 1.0
```
same as above

AlanAboudib replied:
handled!
